kip-0031.md

```diff
@@ -0,0 +1,107 @@
+
+Moreover, token issuers have no clean and defined way to inform wallets when some changes happen with their token (Rebrand, new issuance, ...)
+
+And finally, since Kadena is multichains, wallets need to be informed on how to use the token, regarding chains deployment:
+  - Some tokens may not be available on all chains.
+  - Some tokens may have a single issuance chain where supply is accounted
+
+The proposal is **to create an optional interface that can be implemented by token modules** to automatically inform third parties about their meta-data, and displayable information to users. Wallets, frontends, ... are expected to make local calls to the `kip.fungible-meta` implemented functions.
+
+#### Trustability
+
+Information (especially supply) relies **only** on token issuers and module writers good faith. Clients should not blindly trust them. Sorting legitimate tokens from illegitimate ones is not in the scope of this KIP.
+
+#### Non-Obligability
+For being compliant with this KIP, token issuers have no obligation to provide all data.
+And clients (Wallets, Frontends), have no obligation to show everything.
+
+#### Supply considerations
+
+One important piece of information required by investors is the total and circulating supply.
+Kadena being multichain, when dynamic minting/burning tokens are considered, a new issue arises:
+  - Usually, minting / burning / emission happens on a single chain.
+  - **But other chains are not aware**: they just receive X-chain transfers.
+
+As a consequence, it's highly probable that supply information won't be accurate on most chains. That's why this KIP proposes the concept of **main-chain** => The module itself informs the client where an accurate value of supply amounts can be found.
+
+
+### API
+
+The interface `kip.fungible-meta`, which should be implemented by token modules, provides the following functions.
+
+##### (general-info)
+ -> *object:{kip.fungible-meta.info-schema}*
+
+Return general information about a token:
+  - `symbol` *string*: Token Ticker symbol. Usually a short name in uppercase.
+  - `name` *string*: Token name. Usually a relative short name composed of 1 or 2 words.
+  - `description` *string*: Longer description of the token. Not mandatory
+  - `img` *string* : URL of the graphical representation of the token. Not mandatory
+  - `logo` *string* : URL of a small (res max of 192x192) graphical representation of the token. Not mandatory. Can be the same as `img`
+  - `color` *string*: A CSSv4 compatible color (https://www.w3.org/TR/css-color-4/#typedef-color). Not mandatory. Can be used by wallets or frontends for some graphical displays (charts ...)
+  - `socials` *[object:{kip.fungible-social-schema}]*: A list of tuples (`type` *string*, `url` *string*). The list of allowed types (non limitative): "discord", "twitter", "website", "telegram", "reddit", "github"  
+
+**Note:** All optional fields must use an empty string if the information is not provided.
```

Comment:
which ones are optional? Is this the same as "not mandatory"? Why are `socials` mandatory?